padguide2/padguide2.py

```diff
@@ -2320,43 +2348,75 @@ def find_monster2(self, query):
                 break
         
         # if we don't actually have multiple prefixes, then default to using the regular id lookup
-        if len(query_prefixes) < 2:
+        if len(query_prefixes) < 1:
             return self.find_monster(query)
         
-        matches = set()
+        matches = PotentialMatches()
         
         # first try to get matches from nicknames
         for nickname, m in self.all_entries.items():
             if new_query in nickname:
                 matches.add(m)
-        self.remove_potential_matches_without_all_prefixes(matches, query_prefixes)
+        matches.remove_potential_matches_without_all_prefixes(query_prefixes)
         
         # if we don't have any candidates yet, pick a new method
-        if not len(matches):
+        if not matches.length():

```

Comment:
We talked earlier about returning early if we found a match and I believe you wanted not to do that, to support this feature.

Not sure if I'm misreading it, but I don't see anything in this code enabled by not returning early. Consider changing it.

Reply:
I haven't done anything with it yet, but I still think I will eventually - the biggest thing I want to do is add support for it returning 2 messages, the first is the id request answer, and the 2nd is a list of additional potential matches that weren't as good but were still at least __ good. I'm not sure how likely it is that I actually get around to doing it but I'd rather leave the code easily able to support that in case I do.

Also I wanted to be conservative for now but I think probably it should look up matches on pantheon names & monster names at the same time for each level of checks (exact match on both, then contains match on both, etc) but I wanted to run a check on your historical data before that